### ISSUE TITLE
Text embedding and LLM categories mixed, models not in correct category

### ISSUE THREAD

**Issue by kauttoj** [lmstudio-ai/lmstudio-bug-tracker#66]
See list of my current models in LM Studio. Why are some embedding models included in LLM model category? Which file(s) in LM studio Windows installation I can modify to force a model into specific category? In this case I want to force those two embedding models from LLM to embedding category.
![image](https://github.com/user-attachments/assets/667f6979-36ac-41ea-b9d8-acfbf628d565)

**Comment by yagil** [comment]
Thanks @kauttoj these models are llamas (see screenshots) which makes the app categorize them as LLMs.
I understand you would like to get embeddings output out of these. Other than the categorization in My Models, do they work as you intend?

<img width="714" alt="image" src="https://github.com/user-attachments/assets/c299ba46-6cd3-4aaf-bc3a-14cd07516232">

<img width="705" alt="image" src="https://github.com/user-attachments/assets/8f710104-3358-454a-825c-ffac2ee38d44">

**Comment by Yuki001** [comment]
#290 

i found the code detect model type in lm studio ：

t.doesGgufMetadataIndicateEmbedding=function(e){return["bert","nomic-bert"].includes(e.arch?.toLowerCase())}

u=(0,w.doesGgufMetadataIndicateEmbedding)(i)?g.ModelDomainType.embedding:g.ModelDomainType.llm;

that's not right, qwen2 has embedding model too.

it's better to add the domain type in model settings. so i can change it to right type.

**Comment by yagil** [comment]
We will look into adding a way for you to use LLMs just for embeddings

**Comment by jinyuli** [comment]
any plan to fix it?

**Comment by Capsar** [comment]
This is a much desired feature / bug fix: support for different types of Embedding models.

I am trying to load a model: jina-xlm-roberta-v3 (Jina-Embeddings-v3) but it is not yet supported.

**Comment by zognotadog** [comment]
Also need this fix 👍 , makes LM Studio unusable

**Comment by lox** [comment]
Argh, just spent the morning getting LMStudio setup to handle embeddings only to discover this issue. Most of the modern embeddings models show up as LLMs and don't respond to /v1/embeddings endpoints.

**Comment by yagil** [comment]
Hey @lox @zognotadog which models specifically are not showing as embedding models? HF links would be great

**Comment by lox** [comment]
gte / Qwen 2 family (arch qwen2)
https://huggingface.co/Alibaba-NLP/gte-Qwen2-1.5B-instruct
https://huggingface.co/mradermacher/gte-Qwen2-1.5B-instruct-GGUF (GGUF versions)
https://huggingface.co/tensorblock/gte-Qwen2-7B-instruct-GGUF (GGUF, 7 B)

SFR-Embedding-Mistral family (arch llama)
https://huggingface.co/Salesforce/SFR-Embedding-Mistral
https://huggingface.co/dranger003/SFR-Embedding-Mistral-GGUF (Q4 / Q5 / Q8 GGUF builds for llama.cpp & Ollama)

**Comment by axl84** [comment]
https://huggingface.co/mlx-community/snowflake-arctic-embed-l-v2.0-8bit 
v1.5 as well

**Comment by yagil** [comment]
Implemented in 0.3.16, available as a stable release now: https://lmstudio.ai/download

**Comment by xylagbx** [comment]
The newly updated Qwen3 embedding model is also being identified as an LLM.
https://huggingface.co/Qwen/Qwen3-Embedding-0.6B-GGUF
https://huggingface.co/mlx-community/Qwen3-Embedding-0.6B-4bit-DWQ

**Comment by William-Droin** [comment]
This is still an on going issue, the problem for me is that it makes them unusable as part of the embedding endpoint

**Comment by ErmanGultekin** [comment]
This issue still persists. I can not use the qwen3 embedding models with 0.3.16 version.

**Comment by cageyv** [comment]
In case of someone is still having issues. 
Please update to 0.3.16+ 
Source: https://lmstudio.ai/blog/lmstudio-v0.3.16

```
Added a dropdown in the model editor (⚙️ in My Models) to allow overriding the domain type of a model
This allows you to use an LLM as a text embedding model, or vice versa
```

**Comment by salohcin714** [comment]
Am I crazy or is this option gone now? I've checked on several different models and I am still not seeing the option to override domain.

Currently 0.3.30 on Apple Silicon using MLX

**Comment by codywd** [comment]
@salohcin714 definitely gone. I remember seeing it a little while ago, but it's gone now, at least for MLX. I don't have a GGUF to test with yet. Working to get one now.

**Comment by codywd** [comment]
GGUF still has it. So only affects MLX. Plus, the same Qwen3 Embed 8B model in GGUF was automatically detected as a text embedder. LM Studio 0.3.31b6 on MacOS 26. LM Studio MLX v0.31.0.

**Comment by codywd** [comment]
@yagil

**Comment by zhengyi73** [comment]
Same problem，mlx-community/qwen3-embedding-4b-dwq doesn't have change domain option.

**Comment by kauttoj** [comment]
Wow, ~1.5 years and still without solution. I would love to have tried some Jinaai (jina-embeddings-v5-text-small-retrieval) or Qwen3 (qwen3-embedding-8b) embedding models, but quess not... Apparently there is not even some file-edit hack to enforce model types?
EDIT: The selection is actually there

**Comment by yagil** [comment]
This has been implemented almost a year ago. Go to My Models, pick a model, and edit the domain type, bottom right.

<img width="343" height="242" alt="Image" src="https://github.com/user-attachments/assets/80a5db05-5e15-4bc2-bc0f-84e522503b7b" />

Click on the select element, which will give you

<img width="293" height="96" alt="Image" src="https://github.com/user-attachments/assets/bf7f59bd-0f1b-46f9-9e7d-222fb0e1ba17" />

This is for GGUF only. At this moment, we do not support embedding models for MLX so the only valid domain is LLM.

**Comment by kauttoj** [comment]
> This has been implemented almost a year ago. Go to My Models, pick a model, and edit the domain type, bottom right.
> This is for GGUF only. At this moment, we do not support embedding models for MLX so the only valid domain is LLM.

You are correct, I apologize for my oversight. I didn't notice that selection at the bottom of screen and assumed it was removed, as it was mentioned above. Thanks!
